samcli/lib/telemetry/metric.py

```diff
@@ -476,6 +479,48 @@ def _get_execution_environment(self) -> str:
             return cicd_platform.name
         return "CLI"
 
+    def _get_container_host(self) -> str:
+        """
+        Returns a normalized container engine identifier based on DOCKER_HOST.
+        Maps various DOCKER_HOST patterns to standardized container engine names.
+
+        Examples:
+            - unix:///var/run/docker.sock -> docker
+            - unix://~/.finch/finch.sock -> finch
+            - unix:///run/user/1000/podman/podman.sock -> podman
+            - tcp://localhost:2375 -> tcp-local
+            - (unset) -> docker-default
+            - (other values) -> unknown
+        """
+        if not self._gc.docker_host or not isinstance(self._gc.docker_host, str):
+            return "docker-default"
+
+        docker_host = self._gc.docker_host.lower()
+
+        # Socket mappings
+        socket_map = {"finch.sock": "finch", "podman.sock": "podman", "docker.sock": "docker"}
+
+        # Path-specific mappings (checked first for specificity)
+        path_map = {"colima": "colima", "lima": "lima", "orbstack": "orbstack", ".rd": "rancher-desktop"}
+
```

Comment:
Very minor thing, and it's okay if you don't want to change it, but I feel like if we're checking this first, then we should also **define** this variable first? It's confusing to define `socket_map` and then `path_map`, but then use `path_map` first.